src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/ExceptionUtilities.cs

```diff
@@ -7,9 +7,9 @@ namespace System.Reflection.Internal
     internal static class ExceptionUtilities
     {
         internal static Exception UnexpectedValue(object value) =>
-            new InvalidOperationException($"Unexpected value '{value}' of type '{value?.GetType().FullName ?? "<unknown>"}'");
+            new InvalidOperationException(SR.Format(SR.UnexpectedValue, value, value?.GetType().FullName ?? "<unknown>"));
```

Comment:
STrictly, this should probably be two strings, so that `unknown` is localizable. Eg.,

```
  <data name="UnexpectedValue" xml:space="preserve">
    <value>Unexpected value '{0}' of type '{1}'.</value>
  </data>
  <data name="UnexpectedValueUnknownType" xml:space="preserve">
    <value>Unexpected value '{0}' of unknown type.</value>
  </data>
```